src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityKMSClient.java

```diff
@@ -293,6 +293,74 @@ public static boolean isCiphertext(byte[] bytes) {
         return decryptedDocument;
     }
 
+    /**
+     * Given a map of document IDs to plaintext bytes to encrypt and a map of
+     * document IDs to a fresh DEK, iterate over the DEKs and encrypt the document
+     * at the same key. It is possible that the DEK list is shorter than the
+     * document list in case we had any failures generating new DEKs.
+     */
+    private Map<String, EncryptedDocument> encryptBatchOfDocuments(Map<String, Map<String, byte[]>> documents,
+            Map<String, WrappedDocumentKey> dekList) {
+        return dekList.entrySet().parallelStream().collect(Collectors.toMap(Map.Entry::getKey, dekResult -> {
```

Comment:
It seems like for this (and everywhere else we're using `parallelStream`) we may want to use the `ConcurrentMap` interface and collector to avoid a bunch of intermediate work.